src/classes/BDI_DataImportBatchStatus_TDTM.cls

```diff
@@ -0,0 +1,224 @@
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Batch Data Import
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Trigger Handler for the DataImport SObject to set the batch status
+*/
+public class BDI_DataImportBatchStatus_TDTM extends TDTM_Runnable {
+/*******************************************************************************************************************
+     * @description Trigger Handler on DataImport
+     * @param newSObjects the list of records from trigger new.
+     * @param oldSObjects the list of records from trigger old.
+     * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+     * @param objResult the describe for BatchDataImport
+     * @return dmlWrapper.
+     ********************************************************************************************************************/
+    public override DmlWrapper run(List<SObject> newSObjects, List<SObject> oldSObjects,
+        TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert
+              || triggerAction == TDTM_Runnable.Action.AfterUndelete) {
+            setBatchStatus(newSObjects);
+        }
+
+        if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
+          setBatchStatus(oldSObjects, newSObjects);
+        }
+
+        if (triggerAction == TDTM_Runnable.Action.AfterDelete) {
+          setBatchStatus(oldSObjects);
+        }
+
+        return null;
+    }
+
+    /**
+     * @description updates the batch status on di records insert, delete or undelete
+     * @param diList the old or new list
+     */
+    private void setBatchStatus(List<DataImport__c> diList) {
+      Set<Id> batchIdSet = new Set<Id>();
```

Comment:
This comment applies to the total work here I think - the tab settings look like they are a 2 spaces and I believe our standard is to use 4 spaces?

Reply:
Right, Not sure how I ended up with 2 spaces. I will update all the files.